modular_skyrat/modules/borgs/code/inventory.dm

```diff
@@ -1,13 +1,26 @@
 /mob/living/silicon/robot/unequip_module_from_slot(obj/item/O)
 	..()
-	if(istype(O,/obj/item/gun/energy/laser/cyborg))
+	if(check_held_item_sprites(O))
 		update_icons()
-	else if(istype(O,/obj/item/gun/energy/disabler/cyborg) || istype(O,/obj/item/gun/energy/e_gun/advtaser/cyborg))
-		update_icons() //PUT THE GUN AWAY
 
 /mob/living/silicon/robot/equip_module_to_slot(obj/item/O)
 	..()
-	if(istype(O,/obj/item/gun/energy/laser/cyborg))
-		update_icons() //REEEEEEACH FOR THE SKY
-	if(istype(O,/obj/item/gun/energy/disabler/cyborg) || istype(O,/obj/item/gun/energy/e_gun/advtaser/cyborg))
+	if(check_held_item_sprites(O))
 		update_icons()
+
+/mob/living/silicon/robot/proc/check_held_item_sprites(obj/item/O)
```

Comment:
This, and possibly the list following it, needs comments explaining what it does- it's not clear to me.

Reply:
Or the proc should be renamed